hcipy/util/finite_difference.py

```diff
@@ -18,9 +20,9 @@ def generate_convolution_matrix(grid, kernel):
 	array_like
 		The matrix that applies the convolution.
 	'''
-	num_x = kernel.shape[1]
-	num_y = kernel.shape[0]
-
+	num_x = kernel.grid.shape[1]
+	num_y = kernel.grid.shape[0]
+	
 	yy, xx = np.meshgrid(np.arange(num_y), np.arange(num_y))
```

Comment:
Why use `np.meshgrid()` when you can get the same values from `kernel.grid`?

Reply:
kernel.grid returns the x and y values in physical units. I need to have the values in index units to make sure the kernel is added in the correct places in the sparse convolution matrix.

Reply:
`np.indices()` then?

Reply:
yeah that is a nicer function. I also changed the variables names to better reflect what they are.